raft/handler.go

```diff
@@ -120,11 +121,10 @@ func (h *Handler) serveHeartbeat(w http.ResponseWriter, r *http.Request) {
 	}
 
 	// Execute heartbeat on the log.
-	currentIndex, currentTerm, err := h.Log.Heartbeat(term, commitIndex, leaderID)
+	currentIndex, err := h.Log.Heartbeat(term, commitIndex, leaderID)
 
 	// Return current term and index.
 	w.Header().Set("X-Raft-Index", strconv.FormatUint(currentIndex, 10))
-	w.Header().Set("X-Raft-Term", strconv.FormatUint(currentTerm, 10))
```

Comment:
It makes the code a lot cleaner if we only depose a leader through heartbeats and not on every single possible RPC interaction.